frontend/src/employee-frontend/components/person-profile/PersonFinanceNotesAndMessages.tsx

```diff
@@ -0,0 +1,271 @@
+// SPDX-FileCopyrightText: 2017-2025 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import React, { useContext, useState } from 'react'
+import styled from 'styled-components'
+
+import { FinanceNoteId, PersonId } from 'lib-common/generated/api-types/shared'
+import { cancelMutation, useQueryResult } from 'lib-common/query'
+import AddButton from 'lib-components/atoms/buttons/AddButton'
+import { Button } from 'lib-components/atoms/buttons/Button'
+import { IconOnlyButton } from 'lib-components/atoms/buttons/IconOnlyButton'
+import { MutateButton } from 'lib-components/atoms/buttons/MutateButton'
+import TextArea from 'lib-components/atoms/form/TextArea'
+import {
+  CollapsibleContentArea,
+  ContentArea
+} from 'lib-components/layout/Container'
+import {
+  FixedSpaceColumn,
+  FixedSpaceRow
+} from 'lib-components/layout/flex-helpers'
+import { MutateFormModal } from 'lib-components/molecules/modals/FormModal'
+import { H2, Label, Light } from 'lib-components/typography'
+import { Gap } from 'lib-components/white-space'
+import { faPen, faTrash } from 'lib-icons'
+import { faQuestion } from 'lib-icons'
+
+import { useTranslation } from '../../state/i18n'
+import { PersonContext } from '../../state/person'
+import { UIContext } from '../../state/ui'
+import { formatParagraphs } from '../../utils/html-utils'
+import { renderResult } from '../async-rendering'
+import { FlexRow } from '../common/styled/containers'
+
+import {
+  createFinanceNoteMutation,
+  deleteFinanceNoteMutation,
+  financeNotesQuery,
+  updateFinanceNoteMutation
+} from './queries'
+
+interface Props {
+  id: PersonId
+  open: boolean
+}
+
+export default React.memo(function PersonFinanceNotesAndMessages({
+  id,
+  open: startOpen
+}: Props) {
+  const { i18n } = useTranslation()
+  const { permittedActions } = useContext(PersonContext)
+  const { uiMode, toggleUiMode, clearUiMode } = useContext(UIContext)
+  const [open, setIsOpen] = useState(startOpen)
+  const financeNotes = useQueryResult(financeNotesQuery({ personId: id }))
+  const [text, setText] = useState<string>('')
+  const [confirmDelete, setConfirmDelete] = useState<FinanceNoteId>()
+
+  return (
+    <>
+      {uiMode.startsWith('delete-finance-note') && (
+        <MutateFormModal
```

Comment:
This is fine too, but `ConfirmedMutation` component would achieve the same with less code